### PR TITLE
Tools: Add symbol converter

### PR DESCRIPTION
**Pull request by IoannisP-ITENG**
# Tools: Add symbol converter

# Description
Adds a kicad symbol library to faebryk component library converter.

# Checklist

Please read and execute the following:

- [x] My code follows the [coding guidelines](/faebryk/faebryk/blob/main/docs/CODING_GUIDELINES.md) of this project
- [x] My PR title is following the [contribution guidelines](/faebryk/faebryk/blob/main/docs/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I ran [Black](../docs/CONTRIBUTING.md#creating-a-pull-request) to format my code

#### Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](/faebryk/faebryk/blob/main/docs/CODE_OF_CONDUCT.md):

- [x] I agree to follow this project's [Code of Conduct](/faebryk/faebryk/blob/main/docs/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 networkx
 numpy
-matplotlib
\ No newline at end of file
+matplotlib
+sexpdata
```
Comment (ruben-iteng) [inline requirements.txt:4]: Missing newline at end of file

File: tools/symbol_converter/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# KiCAD symbol converter
+
+This tool converts KiCAD6 symbol libraries to a faebryk component library.
+
+## Usage
```
Comment (ruben-iteng) [inline tools/symbol_converter/README.md:5]: Add newline after chapter title

File: tools/symbol_converter/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+```
```
Comment (ruben-iteng) [inline tools/symbol_converter/README.md:6]: ` ```bash`

File: tools/symbol_converter/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+python3 main.py <path_to_kicad_library>.kicad_sym > <name_of_generated_lib>.py
+```
```
Comment (ruben-iteng) [inline tools/symbol_converter/README.md:8]: Missing newline at end of file

File: tools/symbol_converter/main.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        return self
+
+    def __init__(self):
+        super().__init__()
+
+        self._setup_interfaces()
+    """
+    return template
+
+logger.info("hi")
```
Comment (ruben-iteng) [inline tools/symbol_converter/main.py:109]: hi

File: tools/symbol_converter/main.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+logger.info("Parsing & Converting...")
+
+if len(sys.argv) == 1:
+    filepath = "/usr/share/kicad/symbols/Device.kicad_sym"
```
Comment (ruben-iteng) [inline tools/symbol_converter/main.py:113]: fixed path

Comment (IoannisP-ITENG) [inline tools/symbol_converter/main.py:113]: remove this

**Review by ruben-iteng — CHANGES_REQUESTED** [review]

**Review by ruben-iteng — APPROVED** [review]
Best faebryk tool there is!

File: tools/symbol_converter/main.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        {traits_exp}
+        return
+
+    def _setup_interfaces(self):
+        self.IFs.add_all(times({unnamed_if_cnt}, Electrical))
+        {ifs_exp}
+
+    def __new__(cls, *args, **kwargs):
+        self = super().__new__(cls)
+        self._setup_traits()
```
Comment (ruben-iteng) [inline tools/symbol_converter/main.py:164]: `self._setup_traits()` should be moved to under `self._setup_interfaces()` (line 170)

**Review by ruben-iteng — CHANGES_REQUESTED** [review]

**Comment by IoannisP-ITENG** [comment]
superseded

**Pull request closed without merging**